### ISSUE TITLE
Nm90 Script

### ISSUE THREAD

**Issue by Ramkm77** [Masuzu/ZooeyBot#312]
Thinking to make 1 for myself . But if any coming soon that's gonna save my time. 
Thx

**Comment by rabbidluigi** [comment]
Who are you using? I can't work out how to script in Korwa >.<

**Comment by unoplz** [comment]
if characters["コルワ"]:HasStatusEffect("1534") and characters["コルワ"]:HasStatusEffect("1535") and num_korwa_fils >= 8 then
  EnableChargeAttack()
end

if num_korwa_fils >= 9 then
  if characters["コルワ"].skill_2_available then
    characters["コルワ"]:UseSkill(2)
  end
end
if num_korwa_fils >= 7 and characters["コルワ"]:HasStatusEffect("1534") then
  if characters["コルワ"].skill_1_available then
    characters["コルワ"]:UseSkill(1)
  end
end

I cant figure out how to stop using korwa skill2 when there's buff skill 2...plz help,otherwise this should work pretty well

**Comment by Ramkm77** [comment]
Got all But no gw chars . Can fix the script if don't have the char .

**Comment by tleytek** [comment]
korwa status effect scripts are not going to work in multi battles in 1.22.0

**Comment by Masuzu** [comment]
The status effects will work if you have their ID and use `:HasStatusEffect("ID of the effect")`.
You can also retrieve the number of Korwa's fil with the global variable `num_korwa_fils`.
https://github.com/Masuzu/ZooeyBot/wiki/Lua-Scripts-(playing-with-status-effects)

This bug https://github.com/Masuzu/ZooeyBot/issues/307 is unrelated to this feature.

**Comment by unoplz** [comment]
My script work for me,should work for u guys by changing characters["コルワ"] to characters["korwa"]...
And how can i use skill(2)only when theres no Esprit buff on korwa?

**Comment by Masuzu** [comment]
You can check whether there is no esprit buff on Korwa as follows:
```
if characters["Korwa"] ~= nil and not characters["Korwa"]:HasActiveEffect(1535) then
  characters["Korwa"]:UseSkill(2)
end
```
(please refer to https://github.com/Masuzu/ZooeyBot/wiki/Lua-Scripts-(character-specific) for some status effect IDs)

On my side I use this script to clear consistently the NM 90 mode in 6-7 turns (outside of Assault time) https://github.com/Masuzu/ZooeyBot/blob/master/Scripts/earch_gw_september_2017.lua

**Comment by tleytek** [comment]
I believe bug #307 does relate to the feature HasActiveEffect because it requires RetrieveStatusEffects = true, which doesn't seem to be functioning correctly when turned on and in a Multi Battle. Therefore HasActiveEffect doesn't work at all in Multi Battles.

Here is what I tested in a Trial Battle and had work correctly, but then didn't in EX+.

```
if turn == 1 then
  booleans["cycle incomplete"] = true
  DisableChargeAttack()
end

if num_korwa_fils == 10 and booleans["cycle incomplete"] then
  characters["Korwa"]:UseSkill(2)
end
  
if num_korwa_fils == 10 and characters["Korwa"]:HasStatusEffect("1534") and booleans["cycle incomplete"] then
  characters["Korwa"]:UseSkill(1)
  EnableChargeAttack()
  booleans["cycle incomplete"] = false
end
  
if num_korwa_fils == 10 and characters["Korwa"]:HasStatusEffect("1534") and characters["Korwa"]:HasStatusEffect("1535") then
  EnableChargeAttack()
end

if num_korwa_fils < 10 then
  DisableChargeAttack()
end
```

**Comment by Masuzu** [comment]
Oh sorry, my bad, indeed.
After double checking, setting RetrieveStatusEffects = false disables the use of status effects in scripts, while it should not!
I will release a fix to make the use of status effects in Lua scripts independent from the parameter RetrieveStatusEffects. RetrieveStatusEffects was intended in the first place to create a dictionary of status effects <=> name (as seen in the status_effects.txt file generated when RetrieveStatusEffects = true), at the cost of a longer processing time per turn.

**Comment by Masuzu** [comment]
@tleytek This is now addressed is the newly released version 1.22.2.
